pilot/pkg/model/sidecar.go

```diff
@@ -929,19 +929,20 @@ func (sc *SidecarScope) appendSidecarServices(servicesAdded map[host.Name]sideca
 		sc.servicesByHostname[s.Hostname] = s
 	} else {
 		existing := foundSvc.svc
-		// We donot merge k8s service with any other services from other registries
-		if existing.Attributes.ServiceRegistry == provider.Kubernetes {
-			log.Debugf("Service %s/%s from registry %s ignored by %s/%s/%s", s.Attributes.Namespace, s.Hostname, s.Attributes.ServiceRegistry,
+		// We do not merge k8s service with any other services from other registries
+		if existing.Attributes.ServiceRegistry == provider.Kubernetes && s.Attributes.ServiceRegistry != provider.Kubernetes {
```

Comment:
how could this occur two services in k8s have same fqdn

Reply:
Please try with this sidecar https://github.com/istio/istio/pull/50840#issuecomment-2095169376

Reply:
I see how this occur, it is the egress listener with generate intermeditae svc with only one port from the original k8s svc.

Reply:
Yes and we need to merge at that end